src/components/table/column-picker/column-picker.component.ts

```diff
@@ -41,39 +70,180 @@ export class ColumnPickerComponent {
+    /** Build the heirarchy of the deselect tree */
+    ngOnInit(): void {
+        let treeData: ColumnPickerTreeNode[] = [];
 
-        // add each item to the selected columns list
-        columns.forEach(column => this.selected = [...this.selected, column]);
+        // combine select and deselect into one list
+        this.deselected = this.deselected.concat(this.selected);
 
-        // remove each item from the deselected columns list
-        this.deselected = this.deselected.filter(column => columns.indexOf(column) === -1);
+        const groupedItems: ColumnPickerValue[] = this.deselected.filter(column => this.isColumnPickerItem(column) && (column as ColumnPickerGroupItem).group !== null);
+        const ungroupedItems = this.deselected.filter(column => groupedItems.indexOf(column) === -1);
+        groupedItems.sort((a: ColumnPickerGroupItem, b: ColumnPickerGroupItem) => {
+            // sort by group first
+            if (a.group > b.group) { return -1; }
+            if (a.group < b.group) { return 1; }
+
+            // sort by name after
+            if (a.name > b.name) { return 1; }
+            if (a.name < b.name) { return -1; }
+        });
+
+        ungroupedItems.sort((a: ColumnPickerValue, b: ColumnPickerValue) => {
+            // get names and sort
+            const aName = this.getColumnName(a);
+            const bName = this.getColumnName(b);
+            if (aName > bName) { return 1; }
+            if (aName < bName) { return -1; }
+        });
+
+        let currentGroup: string = null;
+
+        // create grouped items
+        groupedItems.forEach((column: ColumnPickerGroupItem) => {
+            // if new group create top level item
+            if (!currentGroup || column.group !== currentGroup) {
+                currentGroup = column.group;
+
+                // check if settings present for the current group
+                const groupSettings = this.groupSettings.find(setting => setting.group === column.group);
+                const isExpanded = groupSettings && groupSettings.initiallyExpanded || false;
+
+                treeData.push({
+                    name: column.group,
+                    level: 0,
+                    expandable: true,
+                    isExpanded
+                });
+            }
+
+            treeData.push({
+                name: column.name,
+                level: 1,
+                expandable: false
+            });
+        });
+
+        // create ungrouped items
+        ungroupedItems.forEach((column: ColumnPickerValue) => {
+            treeData.push({
+                name: this.getColumnName(column),
+                level: 0,
+                expandable: false
+            });
+        });
+
+        this.treeData = treeData;
+        this.treeDataSource = new ArrayDataSource(treeData);
+    }
+
+    ngAfterViewInit(): void {
+        this._focusKeyManager = new FocusKeyManager(this.treeNodes).withVerticalOrientation();
```

Comment:
I believe you mentioned you were having some issues with keyboard control on standup? You might be able to simplify this by using our `uxTabbableList` directive?